central/telemetry/billingmetrics/scheduler.go

```diff
@@ -30,15 +31,29 @@ func gather() {
 	previousMetrics = newMetrics
 }
 
+// GetCurrent returns the total values of last known metrics reported by known
+// clusters.
+func GetCurrent(ctx context.Context) *clustermetrics.BillingMetrics {
+	ids, err := getClusterIDs(ctx)
+	if err != nil {
+		log.Debug("Failed to get cluster IDs for current billing metrics: ", err)
+		return nil
+	}
+	log.Debugf("Cutting billing metrics for %d clusters: %v", len(ids), ids)
+	return clustermetrics.FilterCurrent(ids)
```

Comment:
I think it's a bit unfortunate that a function called `Get...` is neither idempotent nor immutable. Perhaps we should separate the pruning from the get, and pass the cluster IDs as input?

Reply:
On the other hand `Current` is never immutable.